design-proposals/bound-token-for-vnc.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+
+Kubernetes API server can create 
+[tokens](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens)
+bound to a service account. They are mainly used in services running inside the cluster to authenticate
+between each other. 
+
+This document describes how to create a token that can be used to authenticate with the API server
+to only access VNC endpoint of a singe VM.
+
+## Motivation
+
+An administrator may want to give VNC access to a client application without giving it permissions to access
+anything else in the cluster. The VNC client could be pointed to VMI `/vnc` subresource and use
+the token to authenticate. 
+
+## Goals
+
+- A temporary token can be created to access VNC.
+- It can only be used with one specific VM.
+- It is valid only for a specific duration.
+
+## User Stories
+
+- A VM owner wants to give access to VNC to a third party application without giving it permissions to modify the VM.
+
+## Repos
+
+- https://github.com/kubevirt/kubevirt
+
+# Design
+
+An API subresource under `virtualmachineinstance` , `/vnc-token` is used to get a token for VNC.
+
+The token will refer to a `ServiceAccount` that only has permissions to access the `/vnc` subresource on
+one specific VM.
+
+
+## API Examples
+
+Getting the VNC token:
+```
+GET /apis/subresources.kubevirt.io/v1/namespaces/${VMI_NAMESPACE}/virtualmachineinstances/${VMI_NAME}/vnc-token
+```
+
+Parameters:
+- `duration` - Time duration while the token is valid.
+
```

Comment:
Can you include an example response here?

Reply:
Done